kinto/views/records.py

```diff
@@ -55,23 +53,32 @@ def process_record(self, new, old=None):
         if not schema or not asbool(settings.get(schema_validation)):
             return new
 
-        collection_timestamp = self._collection[self.model.modified_field]
-
+        # Remove internal and auto-assigned fields from schema and record.
+        internal_fields = (self.model.id_field,
+                           self.model.modified_field,
+                           self.schema_field,
+                           self.model.permissions_field)
+        required_fields = [f for f in schema.get('required', []) if f not in internal_fields]
+        if required_fields:
+            schema = {**schema, 'required': required_fields}
+        else:
+            schema = {f: v for f, v in new.items() if f != 'required'}
```

Comment:
I agree with @glasserc that we should do all this at save time, rejecting the schema or changing it silently if it has internal fields in it.